profiling/src/profile/mod.rs

```diff
@@ -1155,6 +1320,1224 @@ mod api_test {
         profile.add(sample).unwrap_err();
     }
 
+    #[test]
+    fn test_no_upscaling_if_no_rules() {
+        let sample_types = vec![
+            api::ValueType {
+                r#type: "samples",
+                unit: "count",
+            },
+            api::ValueType {
+                r#type: "wall-time",
+                unit: "nanoseconds",
+            },
+        ];
+
+        let mut profile: Profile = Profile::builder().sample_types(sample_types).build();
+
+        let id_label = api::Label {
+            key: "my label",
+            str: Some("coco"),
+            num: 10,
+            num_unit: None,
+        };
+
+        let sample1 = api::Sample {
+            locations: vec![],
+            values: vec![1, 10000],
+            labels: vec![id_label],
+        };
+
+        profile.add(sample1).expect("add to success");
+
+        let serialized_profile = pprof::Profile::try_from(&profile).unwrap();
+
+        let first = serialized_profile.samples.get(0).expect("one sample");
+
+        assert_eq!(first.values[0], 1);
+        assert_eq!(first.values[1], 10000);
+    }
+
+    #[test]
+    fn test_upscaling_by_value_a_zero_value() {
+        let sample_types = vec![
+            api::ValueType {
+                r#type: "samples",
+                unit: "count",
+            },
+            api::ValueType {
+                r#type: "wall-time",
+                unit: "nanoseconds",
+            },
+            api::ValueType {
+                r#type: "cpu-time",
+                unit: "nanoseconds",
+            },
+        ];
+
+        let mut profile: Profile = Profile::builder().sample_types(sample_types).build();
+
+        let sample1 = api::Sample {
+            locations: vec![],
+            values: vec![0, 10000, 42],
+            labels: vec![],
+        };
+
+        profile.add(sample1).expect("add to success");
+
+        let upscaling_info = UpscalingInfo::Proportional {
+            total_sampled: 2,
+            total_real: 4,
+        };
+        let values_offset: Vec<usize> = vec![0];
+        profile
+            .add_upscaling_rules(values_offset.as_slice(), "", "", upscaling_info)
+            .expect("Rule added");
+
+        let serialized_profile = pprof::Profile::try_from(&profile).unwrap();
+
+        let first = serialized_profile.samples.get(0).expect("one sample");
+
+        assert_eq!(first.values[0], 0);
+        assert_eq!(first.values[1], 10000);
+        assert_eq!(first.values[2], 42);
```

Comment:
Minor: Is it possible to assert on all values at once? E.g. `assert_eq!(first.values, vec![0, 10000, 42])` it would also save some of the repetition.

Reply:
good point, I did not know